src/lib/draw-and-pick.js

```diff
@@ -136,16 +136,21 @@ export function pickLayers(gl, {
 
     affectedLayers.forEach(layer => {
       let info = Object.assign({}, baseInfo);
-      info.layer = layer;
 
       if (layer === pickedLayer) {
         info.color = pickedColor;
         info.index = pickedObjectIndex;
         info.picked = true;
       }
 
-      // Let layers populate its own info object
-      info = layer.pickLayer({info, mode});
+      // walk up the composite chain and find the owner of the event
+      // sublayers are never directly exposed to the user
+      while (layer && info) {
+        info.layer = layer;
+        // Let layers populate its own info object
+        info = layer.pickLayer({info, mode});
```

Comment:
So all layers in the chain overwrite the same info object? I can see the pros (in terms of simplicity) but also cons with this (inadvertent clobbering of important fields, confusing mix of residual information from topmost and deeper layers).
Just a thought: Do we need more documentation on the pick info object to make the semantics here clearer to users and layer writers?